diff --git a/build/shared.mk b/build/shared.mk
index 1e1afbd..4ba39c6 100644
--- a/build/shared.mk
+++ b/build/shared.mk
@@ -50,8 +50,8 @@ ifeq ($(CONFIG_DEBUG),y)
 FLAGS += -DDEBUG -rdynamic -O0 -ggdb -mssse3 -muintr
 LDFLAGS += -rdynamic
 else
-# FLAGS += -DNDEBUG -O3 -mssse3
-FLAGS += -DNDEBUG -O3 -mavx512f
+FLAGS += -DNDEBUG -O3 -mssse3
+# FLAGS += -DNDEBUG -O3 -mavx512f
 # FLAGS += -DNDEBUG -O3 -mavx2
 ifeq ($(CONFIG_OPTIMIZE),y)
 FLAGS += -march=native -flto -ffast-math
diff --git a/iokernel/dpdk.c b/iokernel/dpdk.c
index cf7d46d..2c19567 100644
--- a/iokernel/dpdk.c
+++ b/iokernel/dpdk.c
@@ -280,7 +280,7 @@ int dpdk_init(void)
 int dpdk_late_init(void)
 {
 	/* initialize port */
-	dp.port = 0;
+	dp.port = 1;
 	if (dpdk_port_init(dp.port, dp.rx_mbuf_pool) != 0) {
 		log_err("dpdk: cannot init port %"PRIu8 "\n", dp.port);
 		return -1;
