diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1bdc506..0c75cfd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,7 +67,7 @@ if(UNIX)  # APPLE, LINUX, FREE_BSD
     set (PCM_COMMON_FLAGS "-Wno-unknown-pragmas -DCMAKE_INSTALL_PREFIX=\"${CMAKE_INSTALL_PREFIX}\"")
 
     if(LINUX)
-       set (PCM_COMMON_FLAGS "${PCM_COMMON_FLAGS} -Wextra -DPCM_USE_PERF")
+       set (PCM_COMMON_FLAGS "${PCM_COMMON_FLAGS} -Wextra -DPCM_USE_PCI_MM_LINUX -DPCM_SILENT")
        if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
          set (PCM_COMMON_FLAGS "${PCM_COMMON_FLAGS} -Wl,-z,now")
        endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 79458f0..2c3cacb 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -8,7 +8,7 @@ set(PROJECT_NAMES pcm pcm-numa pcm-latency pcm-power pcm-msr pcm-memory pcm-tsx
 
 set(MINIMUM_OPENSSL_VERSION 1.1.1)
 
-file(GLOB COMMON_SOURCES pcm-accel-common.cpp msr.cpp cpucounters.cpp pci.cpp mmio.cpp tpmi.cpp pmt.cpp bw.cpp utils.cpp topology.cpp debug.cpp threadpool.cpp uncore_pmu_discovery.cpp pcm-iio-pmu.cpp pcm-iio-topology.cpp lspci.cpp dashboard.cpp ${PCM_PUGIXML_CPP})
+file(GLOB COMMON_SOURCES pcm-accel-common.cpp msr.cpp cpucounters.cpp pci.cpp mmio.cpp tpmi.cpp pmt.cpp bw.cpp utils.cpp topology.cpp debug.cpp threadpool.cpp uncore_pmu_discovery.cpp pcm-iio-pmu.cpp pcm-iio-topology.cpp lspci.cpp dashboard.cpp pcm-iok.cpp ${PCM_PUGIXML_CPP})
 
 if (NOT APPLE)
   file(GLOB UNIX_SOURCES resctrl.cpp)
diff --git a/src/pcm-iok.cpp b/src/pcm-iok.cpp
new file mode 100644
index 0000000..3529355
--- /dev/null
+++ b/src/pcm-iok.cpp
@@ -0,0 +1,40 @@
+#include "cpucounters.h"
+
+using namespace pcm;
+
+extern "C" {
+
+    bool init = false;
+    PCM *p_pcm;
+    ServerUncorePMUs *p_unc;
+
+	uint32_t pcm_iok_get_cas_count(uint32 channel)
+	{
+        uint64_t reads = p_unc->getMCCounter(
+            channel, ServerUncorePMUs::EventPosition::READ);
+        uint64_t writes = p_unc->getMCCounter(
+            channel, ServerUncorePMUs::EventPosition::WRITE);
+        return reads + writes;
+	}
+
+	uint32_t pcm_iok_get_active_channel_count(void)
+	{
+		uint32_t count = 0;
+		for (uint32_t i = 0; i < p_unc->getNumMCChannels(); i++)
+			count += pcm_iok_get_cas_count(i) != 0;
+		return count;
+	}
+
+	int pcm_iok_init(int socket)
+	{
+        if (init) {
+            return 0;
+        }
+
+		p_pcm = PCM::getInstance();
+        p_unc = new pcm::ServerUncorePMUs(socket, p_pcm);
+        p_unc->program();
+        init = true;
+		return 0;
+	}
+}
